Core/Src/main.c

```diff
@@ -69,41 +69,48 @@ static void MX_WWDG_Init(void);
 /* USER CODE BEGIN 0 */
 volatile uint64_t TxData[3];
 uint32_t bit = 0;
-
 uint64_t data = 0;
 uint64_t crc_key = 0xD;
 
 typedef enum{
-  BUTTON_1,
-  BUTTON_2,
-  BUTTON_3,
-  BUTTON_4,
-  //could track up to 32 buttons
-  NONE_B
+	BUTTON_1,
+	BUTTON_2,
+	BUTTON_3,
+	BUTTON_4,
+	//could track up to 32 buttons
+	NONE_B
 }button_state;
 
 button_state button = NONE_B;
 GPIO_PinState state;
+uint32_t button_flag = 0;
+
+uint32_t previousMillis = 0;
```

Comment:
You should keep it consistent with variable casing, so it should 
```suggestion
uint32_t previous_millis = 0;
```

Reply:
Great point, I will change those now